### PR TITLE
Fix bugs in issue dashboard stats

### PR DESCRIPTION
**Pull request by ethantkoenig**
Fixes the following bugs in the issue dashboard:
* Previously, the "In your Repositories", "Assigned to you" and "Created to you" counts would show the stats for open issues, even if closed issues were selected
* Previously, the "In your Repositories" count would be incorrect when a repo that is not yours is selected (the count should be 0, because of all the issues currently in view, none are in repos that are yours)
* Previously, simultaneously selecting "In your Repositories" and a repo that is not yours would result in all the issues in that repo being displayed (no issues should be displayed, since none of the issues in that repo are in repos that are yours)

Also fixes ignored errors in `GetUserIssueStats(..)`, and adds unit tests for affected functionality.

### CODE REVIEW

File: routers/user/home.go [hunk 1]
```diff
@@ -231,21 +231,30 @@ func Issues(ctx *context.Context) {
...
 
+	if repoID > 0 {
+		opts.RepoIDs = []int64{repoID}
+	}
+
 	switch filterMode {
 	case models.FilterModeAll:
-		opts.RepoIDs = userRepoIDs
+		if repoID > 0 {
+			if !com.IsSliceContainsInt64(userRepoIDs, repoID) {
```
Comment (lunny) [inline routers/user/home.go:252]: com is not import

**Review by lunny — CHANGES_REQUESTED** [review]

**Comment by lunny** [comment]
And I didn't find any change on issues page on my local tests.

File: routers/user/home.go (hunk 1 above)
Comment (ethantkoenig) [reply routers/user/home.go:252]: fixed

**Comment by ethantkoenig** [comment]
@lunny Did you try following the scenarios I listed in the PR description? I see a difference in those scenarios.

**Comment by ethantkoenig** [comment]
@lunny Friendly ping

File: routers/user/home.go (hunk 1 above)
Comment (lunny) [reply routers/user/home.go:252]: It seems not?

Comment (ethantkoenig) [reply routers/user/home.go:252]: Strange, fixed now

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
LGTM

**Comment by lafriks** [comment]
LGTM

**Comment by lafriks** [comment]
Oh these random fails :( can you force push?

**Comment by ethantkoenig** [comment]
@lafriks Do you still need a force push?

**Comment by lafriks** [comment]
@ethantkoenig no, all good :)

**Pull request merged**